### ISSUE TITLE
[VSCode] Bug in Visual Studio Code Terminal

### ISSUE THREAD

**Issue by bythckr** [Powerlevel9k/powerlevel9k#1350]
I am on Mac OS Mojave, I am using oh-my-zsh+iTerm2+[powerlevel9k+awesome-patched](https://github.com/Powerlevel9k/powerlevel9k/wiki/Install-Instructions#option-3-install-awesome-powerline-fonts).

But now when I am trying to use the terminal, its having issue with the [emojis/unicode (the branch shape, what is the right term?)](https://camo.githubusercontent.com/31da002de611cfef95f6daaa8b1baedef4079703/687474703a2f2f6268696c6275726e2e6f72672f636f6e74656e742f696d616765732f323031352f30312f706c396b2d696d70726f7665642e706e67).

Under the understanding (I might be wrong), that VSCode's terminal is reading the `.zshrc`. I am assuming that it read `POWERLEVEL9K_MODE='awesome-patched'`. I set the font to `"terminal.integrated.fontFamily": "'SourceCodePro+Powerline+Awesome+Regular',",` in `vscode's settings.json`.

My VSCode:
<img width="942" alt="Screen Shot 2019-08-07 at 13 52 21" src="https://user-images.githubusercontent.com/26061980/62613517-a597fb80-b91a-11e9-9edd-99738b1bb914.png">

My VSCode with `"terminal.integrated.fontFamily": "Inconsolata-dz for Powerline",`:
<img width="923" alt="Screen Shot 2019-08-07 at 13 46 21" src="https://user-images.githubusercontent.com/26061980/62613594-d1b37c80-b91a-11e9-9814-3d0768c05c38.png">

My iTerm on 'awesome-patched' mode in .zshrc & SourceCodePro+Powerline+Awesome+Regular in iterm preference:
<img width="934" alt="Screen Shot 2019-08-07 at 13 35 25" src="https://user-images.githubusercontent.com/26061980/62613633-e3951f80-b91a-11e9-8a7f-55cf51fc44e9.png">

[Inconsolata-dz for Powerline was suggested](https://github.com/Powerlevel9k/powerlevel9k/issues/1030).

What else could I try?

**Comment by romkatv** [comment]
> I set the font to `"terminal.integrated.fontFamily": "'SourceCodePro+Powerline+Awesome+Regular',",` in `vscode's settings.json`.

You need to remove that trailing comma from `fontFamily`.

```diff
- "terminal.integrated.fontFamily": "'SourceCodePro+Powerline+Awesome+Regular',",
+ "terminal.integrated.fontFamily": "'SourceCodePro+Powerline+Awesome+Regular'",
```

**Comment by romkatv** [comment]
It's also possible that you need to replace one `+` with space.

```
"terminal.integrated.fontFamily": "'SourceCodePro+Powerline+Awesome Regular'",
```

**Comment by bythckr** [comment]
The trailing comma & the `+` between `Awesome` & `Regular` was the issue. 2 rather silly issues. Thanks a lot.

There is still one more issue. Repeating the command in the output line. Why is that?

**Comment by romkatv** [comment]
> There is still one more issue. Repeating the command in the output line. Why is that?

I don't see anything about repeating the command in the output line in the issue description. Can you elaborate?

**Comment by bythckr** [comment]
VS Code Terminal: (Note the repeating command circled in red)
<img width="934" alt="Screen Shot 2019-08-09 at 00 18 41" src="https://user-images.githubusercontent.com/26061980/62735525-69f55280-ba3c-11e9-8ba7-14e00ea38725.png">

iTerm:
<img width="688" alt="Screen Shot 2019-08-09 at 00 20 20" src="https://user-images.githubusercontent.com/26061980/62735552-7da0b900-ba3c-11e9-8e9d-e0e9326eaa3f.png">

**Comment by romkatv** [comment]
Indeed, that looks pretty bad. Your iTerm also looks kinda broken.

By the way, you are using the original Powerlevel9k, right?

**Comment by bythckr** [comment]
> original Powerlevel9k, right?

I am sorry, I don't understand "original"? I downloaded from https://github.com/Powerlevel9k/powerlevel9k/.

**Comment by romkatv** [comment]
> > original Powerlevel9k, right?
> 
> I am sorry, I don't understand "original"? I downloaded from https://github.com/Powerlevel9k/powerlevel9k/.

Yes, that's the one. I just wanted to make sure we are talking about the same thing.

**Comment by bythckr** [comment]
what do I do?

**Comment by romkatv** [comment]
Maybe some of the powerlevel9k developers or users will help. (I'm neither.)

**Comment by bythckr** [comment]
> (I'm neither.)

I assumed that you are the lead developer. None the less, your support so far has been great. Thanks a lot

**Comment by romkatv** [comment]
@bythckr It appears that this issue is not specific to Powerlevel9k. See https://github.com/romkatv/powerlevel10k/issues/199. Perhaps the resolution on that issue will help you, too.
